Frameworks/Foundation/NSFileManager.mm

```diff
@@ -436,36 +436,48 @@ - (BOOL)copyItemAtPath:(id)srcPath toPath:(id)destPath error:(NSError**)error {
 
     TraceVerbose(TAG, L"Done copying");
 
-    return TRUE;
+    return YES;
 }
 
 /**
- @Status Stub
+ @Status Caveat
+ @Notes does not support file manager move delegates for shouldMoveItem or shouldProceedAfterError.
 */
 - (BOOL)moveItemAtURL:(NSURL*)srcURL toURL:(NSURL*)dstURL error:(NSError**)error {
-    UNIMPLEMENTED();
-
-    return NO;
+    return [self moveItemAtPath:[srcURL path] toPath:[dstURL path] error:error];
 }
 
 /**
  @Status Caveat
- @Notes error parameter is not populated
+ @Notes does not support file manager move delegates for shouldMoveItem or shouldProceedAfterError.
 */
-- (BOOL)moveItemAtPath:(id)srcPath toPath:(id)destPath error:(NSError**)error {
+- (BOOL)moveItemAtPath:(NSString*)srcPath toPath:(NSString*)destPath error:(NSError**)error {
+    THROW_NS_IF_FALSE(E_INVALIDARG, srcPath != nil);
+    THROW_NS_IF_FALSE(E_INVALIDARG, destPath != nil);
+
+    if (![self fileExistsAtPath:srcPath]) {
+        if (error) {
+            *error = [NSError errorWithDomain:NSCocoaErrorDomain code:NSFileNoSuchFileError userInfo:nil];
+        }
+
+        return NO;
+    }
+
     const char* src = [srcPath UTF8String];
     const char* dest = [destPath UTF8String];
 
     TraceVerbose(TAG, L"Moving %hs to %hs", src, dest);
```

Comment:
In the wake of these logs, could you scrub this file for any unnecessary logging? It might also be worth looking into standardizing the error messages while you're in here since you seem to be touching on those a bit.